### PR TITLE
fix #3540

### LINKED ISSUE #3540: [Bug]: copy transform does not overwrite target file
### Apache Hop version?

apache-hop-client 2.7

### Java version?

java-version

### Operating system

Windows

### What happened?

when using the transform "process files" to copy and replace existing fiiles with the same name (liefd001.ild), it does not overwrite the target file although this setting is set. Instead it merges the file content which is not intended. 

screenshot copy bug
screenshot copy bug2

steps in pipeline:
- get file names
- get variables as destination
- regex evaluation to evaluate path
- calculator to concat destination path to get target filename field
- process files to copy files with setting "overwrite target file"

### Issue Priority

Priority: 2

### Issue Component

Component: Transforms

### PR DESCRIPTION
**Pull request by sramazzina**
fix #3540 copy transform does not overwrite target file

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:
- [x] Run `mvn clean install apache-rat:check` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
- [x] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`.
- [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
- [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by sramazzina** [comment]
Basically the overwrite happened without any changes too, I was unable to reproduce the error the reporter experienced. But by performing the test I experienced another issue and that was the reason of the proposed change.

Suppose that we are copying a new text file over an existing one:
- The target file, present in the destination directory, contained the following sample text: THIS IS A SAMPLE
- The file that we are going to copy contains the following text: APACHE HOP
The file that resulted after the copy was something like APACHE HOPSAMPLE. 

Basically, the behavior of the copyFrom API is that it overwrites the file but in terms of bytes. I mean, the first 10 bytes of the target file will be overwritten with the content of the new file (that is 10 byte in size of course). I tried to investigate on the copyFrom API but I wasn0t able to find something related to this behavior. Therefore, for this reason, I decided to delete the target before the copy to be sure that the ovewrite will happen properly.

**Comment by usbrandon** [comment]
Waiting on an update for the case that the file does not exist, but the box is checked.  Making sure a null object is not deleted.

**Comment by sramazzina** [comment]
I'm sorry I wrote about it on mattermost. I will copy the message here:

_"I made some tests this morning and the delete statement do not throws any exception if the target file doesn't exists. I remember I made this change quite some time ago but I pushed only lastly because of the big work released yesterday evening on Static Schema Definitions. Maybe I had already tried the behavior of the delete at that time. In any case it works even if the target file doesn't exists and thanks for pointing me on that. As a developer, tests are always something where I can make things better"_
